provisioner/terraform/install.go

```diff
@@ -49,9 +49,13 @@ func Install(ctx context.Context, log slog.Logger, dir string, wantVersion *vers
 
 	binPath := filepath.Join(dir, product.Terraform.BinaryName())
 
+	hasVersionStr := "nil"
 	hasVersion, err := versionFromBinaryPath(ctx, binPath)
-	if err == nil && hasVersion.Equal(wantVersion) {
-		return binPath, err
+	if err == nil {
+		hasVersionStr = hasVersion.String()
+		if hasVersion.Equal(wantVersion) {
+			return binPath, err
+		}
 	}
```

Comment:
Should this be in a separate PR?

Reply:
It's true that it's a bug fix. Tests weren't exercising the `Debug` log prior to this, so we never caught that this could segfault.

I can strip it out into it's own PR if you like.

Reply:
I'm fine leaving it here!